packages/block/src/block.ts

```diff
@@ -103,6 +105,8 @@ export class Block {
       withdrawals: withdrawalsData,
       executionWitness: executionWitnessData,
     } = blockData
+    this.keccakFunction = this.keccakFunction ?? opts?.common?.customCrypto.keccak256 ?? keccak256
```

Comment:
These assignments shouldn't be done in the static constructors but only one-time in the main constructors from `Block` and `BlockHeader`, otherwise this produces unnecessary code redundancy.

(and does `this` even work here already since we have no object context yet??)

Optimally also *after* the `common` assignment, then the `?` after `common` becomes unnecessary.

Also: the `this.keccakFunction ?? ` should be unnecessary if I am not overlooking something since it should/is by design the first (and only) assignment.

Reply:
(or let me know if there is a reason why this is put here, I was a bit in doubt in between but now again I am not seeing any)

Reply:
(and, to add on this: if there *is* a reason we should really try to remove that reason, since this is atm really not beautiful and this reason might generally also point to some suboptimality in the (existing) code)

Reply:
I am now done with this and think this was really some logic error on your side. Do not see any limitations or things malfunctioning by this.